File: day-7/solution.py
class Solution:
	def amusementPark(self, rides: List[List[int]]) -> bool:
    	keys,prev = {0},0
    	while prev<len(keys):
        	prev=len(keys)
        	keys |= { i for k in keys for i in rides[k] }
    	return prev == len(rides)

File: day-7/solution.java
class Solution {
	public boolean amusementPark(List<List<Integer>> rides) {
    	Stack<Integer> stack = new Stack();
    	boolean[] seen = new boolean[rides.size()];
    	seen[0] = true;
    	stack.push(0);
    	while (!stack.isEmpty()) {
        	int key = stack.pop();
        	for (int newKey : rides.get(key)) {
            	if (!seen[newKey]) {
                	seen[newKey] = true;
                	stack.push(newKey);
            	}
        	}
    	}
    	for (boolean room: seen) {
        	if (room == false) return false;
    	}
    	return true;
	}
}
